services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java

```diff
@@ -126,6 +128,10 @@ private static S3AsyncClient initializeS3AsyncClient(DefaultS3CrtClientBuilder b
             builder.executionInterceptors.forEach(overrideConfigurationBuilder::addExecutionInterceptor);
         }
 
+        if (builder.credentialsProvider == null) {
```

Comment:
Why dont we assign Default here https://github.com/aws/aws-sdk-java-v2/blob/beaa6ef617dd6f81924756f8a10c89cf496685e9/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java#L210 ?
```
        private IdentityProvider<? extends AwsCredentialsIdentity> credentialsProvider = DefaultCredentialsProvider.builder().build();

```

Reply:
IMO builder class should not have any business logic like defining defaults. This is especially true for members that may have resources associated and need to be cleaned up, such as credentials provider